### ISSUE TITLE
Add an audit sub-system to the pbench-server

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#2281]
We really aught to consider adding an audit event, and make an official audit interface that we define as part of the behavior for operations and actions taken for and on behalf of a user.

_Originally posted by @portante in https://github.com/distributed-system-analysis/pbench/pull/2264#discussion_r624051974_

**Comment by dbutenhof** [comment]
Now I can't find it; but some time ago when I started on the Dataset tracking DB, I added some background on audit logging in the context of @riya-17's work on refactoring the service audit task, where it was basically overloading the server-reports index with what were essentially audit records.

Basically, we need a persistent searchable list of events that affect the state of the server resources, following the lifecycle of datasets, users, server components, and anything else. This can be Postgres, Elasticsearch, or something else; but basically needs to provide a list of answers to the "what, when, who, why" questions for everything that changes over time.

We need to define precisely what data to record for each event, the format of the data, and where to keep it. We should investigate audit trail packages to see if we can find something usable off-the-shelf, although it's not incredibly difficult to build.
